annotations-runtime/src/main/java/com/github/dynamicextensionsalfresco/webscripts/AnnotationWebScript.java

```diff
@@ -148,14 +161,65 @@ protected Object invokeUriHandlerMethod(final AnnotationWebScriptRequest request
 	protected void handleUriMethodReturnValue(HandlerMethods handlerMethods, final AnnotationWebScriptRequest request,
                                               final AnnotationWebscriptResponse response, final Object returnValue) throws Exception {
         Resolution resolution = null;
-        if (returnValue instanceof Map) {
+        if (returnValue instanceof Map) { // returning a map will result in response template by default.
             resolution = new TemplateResolution((Map<String, Object>) returnValue);
-        } else if (returnValue instanceof String) {
+        } else if (returnValue instanceof String) { // returning a string will also result in a template response.
             resolution = new TemplateResolution((String)returnValue);
         } else if (returnValue instanceof Resolution) {
             resolution = (Resolution) returnValue;
         }
-        if (this.handlerMethods.useResponseTemplate()) {
+
+        /**
+         * If the method is annotated with {@link org.springframework.web.bind.annotation.ResponseBody}
+         * The response should be serialized automatically by what the request asked in the accept header.
+         * If the accept header is not present, the default format value of the method is used.
+         * If there is no default format available, an exception is thrown.
+         */
+        if (handlerMethods.useResponseBody()){
+            String[] responseTypes = request.getHeaderValues("Accept");
+
+            MediaType acceptResponseType = MediaType.ALL; // default
+
+            if (responseTypes != null){
```

Comment:
This logic needs to be inverted:

- First use the default-format and check if it matches the Accept-header and run this through the message-converters canWrite-method
- If not, look at the values in the Accept-header and try to run those through the canWrite-method